partner_identification_dea/models/res_partner.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    dea_number = fields.Char(
+        string="DEA #",
+    )
+    dea_expired_date = fields.Date(string="DEA Expiration Date")
+    dea_active = fields.Selection([("yes", "Yes"), ("no", "NO")], string="DEA Active")
+
+    medical_license = fields.Char(
+        string="Medical License",
+    )
+    medical_license_expired_date = fields.Date(string="Medical License Expiration Date")
+
+    @api.model
+    def send_expiration_date_notification(self):
+        """Method is used to send Notification Exiration Date of
+        Medical License and DEA Number.
+        """
+        email_dea_template_id = self.env.ref(
+            "partner_identification_dea.email_template_dea_notification",
+            raise_if_not_found=False,
+        )
+        email_medical_template_id = self.env.ref(
+            "partner_identification_dea.email_template_medical_notification",
+            raise_if_not_found=False,
+        )
+        des_partner_ids = self.search(
+            [
+                ("dea_expired_date", "=", fields.Date.today() + relativedelta(days=30)),
+                ("dea_active", "=", "yes"),
+            ]
+        )
+        medical_partner_ids = self.search(
+            [
+                (
+                    "medical_license_expired_date",
+                    "=",
+                    fields.Date.today() + relativedelta(days=30),
+                )
+            ]
+        )
+        for partner in des_partner_ids:
+            email_dea_template_id.send_mail(
+                partner.id,
+                force_send=True,
+            )
+
+        for partner in medical_partner_ids:
+            email_medical_template_id.send_mail(
+                partner.id,
+                force_send=True,
+            )
```

Comment:
I suggest if you would like to have an expiration reminder then you have to improve **partner_identification** module so you will have a generic email sending mechanism

Reply:
+1